packages/protocol/contracts/thirdparty/onchainRA/interfaces/ISigVerifyLib.sol

```diff
@@ -0,0 +1,62 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+interface ISigVerifyLib {
+    enum KeyType {
+        RSA,
+        ECDSA
+    }
+
+    struct PublicKey {
+        KeyType keyType;
+        // If RSA, pubKey = abi.encodePacked(exponent, modulus)
+        // If ECDSA, pubKey = abi.encodePacked(gx, gy)
+        bytes pubKey;
+    }
+
+    enum CertSigAlgorithm {
+        Sha256WithRSAEncryption,
+        Sha1WithRSAEncryption
+    }
+
+    struct Certificate {
+        // Asn.1 DER encoding of the to-be-signed certificate
+        bytes tbsCertificate;
+        PublicKey publicKey;
+        bytes signature;
+        CertSigAlgorithm sigAlg;
+    }
+
+    enum Algorithm {
+        RS256,
+        ES256,
+        RS1
+    }
+
+    function verifyAttStmtSignature(bytes memory tbs, bytes memory signature, PublicKey memory publicKey, Algorithm alg)
```

Comment:
can we have a full name? verifyAttestationStatementSignature? If Stmt is stamement, where is the Statement struct defined, or what's the different between an "Attestation" and an "AttestationStatement"?

Reply:
Need a better name for `tbs`